src/components/App.tsx

```diff
@@ -29,6 +30,20 @@ interface IAppProps extends RouteComponentProps<any> {
 @inject('mapStore', 'loginStore')
 @observer
 class App extends React.Component<IAppProps, IAppState> {
+    async componentWillMount() {
+        const pathName = navigator.getPathName();
+        if (!pathName.includes(SubSites.afterLogin)) {
+            const response = (await ApiClient
+                .postRequest(endpoints.EXISTING_SESSION, {}) as IAuthorizationResponse);
+            const { history } = this.props;
+            if (response.isOk) {
+                this.props.loginStore!.setAuthenticationInfo(response);
+            } else {
+                history.push(SubSites.login);
+            }
+        }
+        this.props.loginStore!.setIsLoginInProgress(false);
```

Comment:
Why is this in store and not state? 
This field is only used in this component, component isolated observables should be in state.